ota-plus-web/app/com/advancedtelematic/auth/oidc/NamespaceProvider.scala

```diff
@@ -22,19 +24,33 @@ class NamespaceFromIdentity extends NamespaceProvider {
 class NamespaceFromUserProfile @Inject()(val conf: Configuration,
                                          val ws: WSClient,
                                          val clientExec: ApiClientExec,
+                                         val oidcGateway: OidcGateway,
                                          val tracer: ZipkinTraceServiceLike)
                                         (implicit ec: ExecutionContext)
   extends NamespaceProvider with ApiClientSupport {
 
+  implicit private val traceData = TraceData(tracer.tracing.tracer().newTrace())
+
+  private val namespaceFromJson: JsValue => Namespace = json => Namespace((json \ "defaultNamespace").as[String])
+
+  private def createUser(accessToken: AccessToken, userId: UserId) =
+    oidcGateway
+      .getUserInfo(accessToken)
+      .flatMap { ic =>
+        userProfileApi
+          .createUser(userId, ic.name, ic.email)
+          .map(namespaceFromJson)
+      }
+
   override def apply(tokens: Tokens): Future[Namespace] = {
-    implicit val traceData = TraceData(tracer.tracing.tracer().newTrace())
-
-    userProfileApi.userOrganizations(tokens.idToken.userId).map {
-      case organizations if organizations.isEmpty =>
-        Namespace.generate
-      case organizations =>
-        organizations.filter(_.isDefault).map(_.namespace).head
-    }
+    val userId = tokens.idToken.userId
+    userProfileApi
+      .getUser(userId)
+      .map(namespaceFromJson)
+      .recoverWith {
+        case RemoteApiError(result, _) if result.header.status == 404 =>
+          createUser(tokens.accessToken, userId)
+      }
```

Comment:
I would suggest to move `.map(namespaceFromJson)` here, make it a last call

Reply:
Done, amended in the same commit.